packages/pages/src/components/hours/hoursTable.tsx

```diff
@@ -0,0 +1,215 @@
+ * @param {String} startOfWeek set the day of the first row of the table
+ * @param {Boolean} collapseDays combine adjacent rows (days) with the same intervals
+ * @param {Function} intervalStringsBuilderFn override rendering for the interval on each table row
+ */
+const HoursTable: React.FC<HoursTableProps> = (props) => {
+  // Use two rendering passes to avoid SSR issues where server & client rendered content is different
+  //  On the first pass, don't render any content in this component, only set `state.isClient`
+  //  On the second pass (After the page has been loaded), render the content
+  // https://reactjs.org/docs/react-dom.html#hydrate
+  const [isClient, setIsClient] = useState(false);
+  useEffect(() => {
+    setIsClient(true);
+  }, []);
+
+  const h = new Hours(props.hours);
+  const now = new Date();
+
+  const dayOfWeekNames = props.dayOfWeekNames
+    ? dayOfWeekNamesToArray(props.dayOfWeekNames)
+    : defaultDayOfWeekNames;
+  const dayOfWeekSortIdx = getSortIdx(props, new Date());
+
+  // Fetch intervals for the next 7 days
+  const allIntervals = h.getIntervalsForNDays(7, now);
+
+  // Split intervals into buckets by day of week
+  let hoursDays: HoursTableDayData[] = [];
+  for (let i = 0; i < 7; i++) {
+    hoursDays.push({
+      dayOfWeek: dayOfWeekNames[i],
+      sortIdx: dayOfWeekSortIdx[i],
+      intervals: allIntervals.filter(
+        (interval) => interval.start.getDay() === i
+      ),
+      isToday: now.getDay() === i,
+    });
+  }
+
+  // Sort the days
+  const sortFn = (day1: HoursTableDayData, day2: HoursTableDayData) => {
+    if (day1.sortIdx === day2.sortIdx) {
+      return 0;
+    }
+    return day1.sortIdx > day2.sortIdx ? 1 : -1;
+  };
+  hoursDays.sort(sortFn);
+
+  // Collapse the days
+  if (props.collapseDays) {
+    hoursDays = collapseDays(hoursDays);
+  }
+
+  return (
+    <>
+      {isClient && (
+        <div className={c("HoursTable", props.className)}>
+          {hoursDays.map((dayData) => {
+            const intervalStringsBuilderFn =
+              props.intervalStringsBuilderFn || defaultIntervalStringsBuilder;
+            const intervalStrings = intervalStringsBuilderFn(
+              dayData,
+              props.timeOptions
+            );
+
+            return (
+              <div
+                className={c("HoursTable-row", { "is-today": dayData.isToday })}
+                key={dayData.sortIdx}
+              >
+                <span className="HoursTable-day">{dayData.dayOfWeek}</span>
+                <span className="HoursTable-intervals">
+                  {intervalStrings.map((intervalString, idx) => (
+                    <span className="HoursTable-interval" key={idx}>
+                      {intervalString}
+                    </span>
+                  ))}
+                </span>
+              </div>
+            );
+          })}
+        </div>
+      )}
+    </>
+  );
+};
+
+export { HoursTable };
```

Comment:
Is this the name you all want to go with vs just calling it Hours? The PMs might have an opinion.

Reply:
Commented in slackhttps://yextops.slack.com/archives/C02TTUT4C5Q/p1660580587284159